coffee_shop_finder.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from Services.user_service import UserService
+from Services.csv_service import CSVService
+from Services.coffee_shop_service import CoffeeShopService
+from Utils.custom_csv_error import CSVServiceError
+
+
+def main():
+    user_args = sys.argv[1:]
+    user_service = UserService(args=user_args)
+    csv_service = CSVService()
+
+    try:
+        user_data = user_service.get_user_data()
+        csv_data = csv_service.get_csv_data(user_data.csv_url)
+    except (IndexError, CSVServiceError) as error:
```

Comment:
Is great that you handled specific errors instead of generic ones,

Why did you choose to go with IndexError instead of a custom error class like the CSVServiceError?

Reply:
I'm expecting index error when parsing user input data, so it's not a CSVServiceError but it could be a UserServiceError.